flake test case: snapshots/devmapper TestSnapshotterSuite/RemoveIntermediateSnapshot

### Description

It is flake test case.

https://github.com/containerd/containerd/actions/runs/3526988813/jobs/5915601489

```
=== Failed
=== FAIL: snapshots/devmapper TestSnapshotterSuite/RemoveIntermediateSnapshot (3.50s)
    log_hook.go:40: time="2022-11-22T21:03:44.910547543Z" level=info msg="initializing pool device \"containerd-snapshotter-suite-pool-733737041\"" func=devmapper.NewPoolDevice file="/home/runner/work/containerd/containerd/snapshots/devmapper/pool_device.go:48"
    log_hook.go:40: time="2022-11-22T21:03:44.925137193Z" level=info msg="using dmsetup:\nLibrary version:   1.02.167 (2019-11-30)\nDriver version:    4.45.0" func=devmapper.NewPoolDevice file="/home/runner/work/containerd/containerd/snapshots/devmapper/pool_device.go:56"
    log_hook.go:40: time="2022-11-22T21:03:44.941321636Z" level=debug msg=prepare func="devmapper.(*Snapshotter).Prepare" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:202" key=/tmp/snapshot-suite-devmapper-889262210/work/base parent=
    log_hook.go:40: time="2022-11-22T21:03:44.942645913Z" level=debug msg="creating new thin device 'containerd-snapshotter-suite-pool-733737041-snap-1'" func="devmapper.(*Snapshotter).createSnapshot" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:409"
    log_hook.go:40: time="2022-11-22T21:03:45.058742177Z" level=debug msg="Creating file system of type: ext4 with options: nodiscard,lazy_itable_init=0,lazy_journal_init=0 for thin device \"containerd-snapshotter-suite-pool-733737041-snap-1\"" func="devmapper.(*Snapshotter).createSnapshot" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:423"
    log_hook.go:40: time="2022-11-22T21:03:45.059046495Z" level=debug msg="mkfs.ext4 -E nodiscard,lazy_itable_init=0,lazy_journal_init=0 /dev/mapper/containerd-snapshotter-suite-pool-733737041-snap-1" func=devmapper.mkfs file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:488"
    log_hook.go:40: time="2022-11-22T21:03:45.081410298Z" level=debug msg="mkfs:\nmke2fs 1.45.5 (07-Jan-2020)\nCreating filesystem with 4096 4k blocks and 4096 inodes\n\nAllocating group tables: 0/1\b\b\b   \b\b\bdone                            \nWriting inode tables: 0/1\b\b\b   \b\b\bdone                            \nCreating journal (1024 blocks): done\nWriting superblocks and filesystem accounting information: 0/1\b\b\b   \b\b\bdone\n\n" func=devmapper.mkfs file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:495"
    log_hook.go:40: time="2022-11-22T21:03:45.131256002Z" level=debug msg=commit func="devmapper.(*Snapshotter).Commit" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:238" key=/tmp/snapshot-suite-devmapper-889262210/work/base name=/tmp/snapshot-suite-devmapper-889262210/work/committed-base
    log_hook.go:40: time="2022-11-22T21:03:45.255775058Z" level=debug msg=prepare func="devmapper.(*Snapshotter).Prepare" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:202" key=/tmp/snapshot-suite-devmapper-889262210/work/intermediate parent=/tmp/snapshot-suite-devmapper-889262210/work/committed-base
    log_hook.go:40: time="2022-11-22T21:03:45.256193182Z" level=debug msg="creating snapshot device 'containerd-snapshotter-suite-pool-733737041-snap-2' from 'containerd-snapshotter-suite-pool-733737041-snap-1' with fsType: 'ext4'" func="devmapper.(*Snapshotter).createSnapshot" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:439"
    log_hook.go:40: time="2022-11-22T21:03:45.25925[406](https://github.com/containerd/containerd/actions/runs/3526988813/jobs/5915601489#step:9:407)0Z" level=debug msg="suspending \"containerd-snapshotter-suite-pool-733737041-snap-1\" before taking its snapshot" func="devmapper.(*PoolDevice).CreateSnapshotDevice" file="/home/runner/work/containerd/containerd/snapshots/devmapper/pool_device.go:352"
time="2022-11-22T21:03:45Z" level=debug msg=close
    log_hook.go:40: time="2022-11-22T21:03:45.516014621Z" level=debug msg=commit func="devmapper.(*Snapshotter).Commit" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:238" key=/tmp/snapshot-suite-devmapper-889262210/work/intermediate name=/tmp/snapshot-suite-devmapper-889262210/work/committed-inter
    log_hook.go:40: time="2022-11-22T21:03:45.601051176Z" level=debug msg=prepare func="devmapper.(*Snapshotter).Prepare" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:202" key=/tmp/snapshot-suite-devmapper-889262210/work/toplayer parent=/tmp/snapshot-suite-devmapper-889262210/work/committed-inter
    log_hook.go:40: time="2022-11-22T21:03:45.601759417Z" level=debug msg="creating snapshot device 'containerd-snapshotter-suite-pool-733737041-snap-3' from 'containerd-snapshotter-suite-pool-733737041-snap-2' with fsType: 'ext4'" func="devmapper.(*Snapshotter).createSnapshot" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:439"
    log_hook.go:40: time="2022-11-22T21:03:45.776547101Z" level=debug msg=remove func="devmapper.(*Snapshotter).Remove" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:295" key=/tmp/snapshot-suite-devmapper-889262210/work/committed-inter
    log_hook.go:40: time="2022-11-22T21:03:45.776858119Z" level=debug msg="snapshotter error" func="devmapper.(*Snapshotter).withTransaction" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:562" error="1 error occurred:\n\t* cannot remove snapshot with child: failed precondition\n\n"
    log_hook.go:40: time="2022-11-22T21:03:45.777099733Z" level=debug msg=remove func="devmapper.(*Snapshotter).Remove" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:295" key=/tmp/snapshot-suite-devmapper-889262210/work/toplayer
    log_hook.go:40: time="2022-11-22T21:03:45.870961502Z" level=debug msg=remove func="devmapper.(*Snapshotter).Remove" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:295" key=/tmp/snapshot-suite-devmapper-889262210/work/committed-inter
    helpers_unix.go:33: unmount /tmp/snapshot-suite-devmapper-889262210/work/base
    log_hook.go:40: time="2022-11-22T21:03:45.884816109Z" level=debug msg=remove func="devmapper.(*Snapshotter).Remove" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:295" key=/tmp/snapshot-suite-devmapper-889262210/work/committed-base
    log_hook.go:40: time="2022-11-22T21:03:46.141003636Z" level=error msg="failed to remove device" func="devmapper.(*Snapshotter).removeDevice" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:311" error="failed to deactivate device \"containerd-snapshotter-suite-pool-733737041-snap-1\": failed to deactivate device: device or resource busy"
    log_hook.go:40: time="2022-11-22T21:03:46.141393959Z" level=debug msg="snapshotter error" func="devmapper.(*Snapshotter).withTransaction" file="/home/runner/work/containerd/containerd/snapshots/devmapper/snapshotter.go:562" error="1 error occurred:\n\t* failed precondition\n\n"
    testsuite.go:556: failed precondition
    helpers.go:66: drwx------       [409](https://github.com/containerd/containerd/actions/runs/3526988813/jobs/5915601489#step:9:410)6 /tmp/snapshot-suite-devmapper-889262210
    helpers.go:66: drwxr-xr-x       4096 /tmp/snapshot-suite-devmapper-889262210/root
    helpers.go:64: -rw-------      32768 /tmp/snapshot-suite-devmapper-889262210/root/containerd-snapshotter-suite-pool-733737041.db [ "\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\xed\xda\f\xed\x02\x00\x00\x00\x00\x10\x00\x00\x00\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\x06\x00\x00\x00\x00\x00\x00\x00" ...]
    helpers.go:64: -rw-------  13[421](https://github.com/containerd/containerd/actions/runs/3526988813/jobs/5915601489#step:9:422)7728 /tmp/snapshot-suite-devmapper-889262210/root/devmapper-snapshotter-tests-1783218348 [ "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" ...]
    helpers.go:64: -rw-------  134217728 /tmp/snapshot-suite-devmapper-889262210/root/devmapper-snapshotter-tests-1918653855 [ "\v~O\x06\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xbaR\x9c\x01\x00\x00\x00\x00\x02\x00\x00\x00\x02\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" ...]
    helpers.go:64: -rw-------      65536 /tmp/snapshot-suite-devmapper-889262210/root/metadata.db [ "\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\xed\xda\f\xed\x02\x00\x00\x00\x00\x10\x00\x00\x00\x00\x00\x00\x06\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\a\x00\x00\x00\x00\x00\x00\x00\r\x00\x00\x00\x00\x00\x00\x00" ...]
    helpers.go:66: drwxr-xr-x       4096 /tmp/snapshot-suite-devmapper-889262210/work
    helpers.go:66: drwxr-xr-x       4096 /tmp/snapshot-suite-devmapper-889262210/work/base
time="2022-11-22T21:03:47Z" level=debug msg=close
    --- FAIL: TestSnapshotterSuite/RemoveIntermediateSnapshot (3.50s)
```

### Steps to reproduce the issue

Unknown.


### Describe the results you received and expected

It should pass.

### What version of containerd are you using?

Main branch

### Any other relevant information

NONE

### Show configuration if it is related to CRI plugin.

NONE

--- a/snapshots/devmapper/pool_device.go
+++ b/snapshots/devmapper/pool_device.go
@@ -88,6 +88,15 @@ func NewPoolDevice(ctx context.Context, config *Config) (*PoolDevice, error) {
 	return poolDevice, nil
 }
 
+func skipRetry(err error) bool {
+	if err == nil {
+		return true // skip retry if no error
+	} else if !errors.Is(err, unix.EBUSY) {
+		return true // skip retry if error is not due to device or resource busy
+	}
+	return false
+}
+
 func retry(ctx context.Context, f func() error) error {
 	var (
 		maxRetries = 100
@@ -97,9 +106,8 @@ func retry(ctx context.Context, f func() error) error {
 
 	for attempt := 1; attempt <= maxRetries; attempt++ {
 		retryErr = f()
-		if retryErr == nil {
-			return nil
-		} else if retryErr != unix.EBUSY {
+
+		if skipRetry(retryErr) {
 			return retryErr
 		}
 

